File: R/Scripts/statistics/boxplot_and_violin_over_category.R
airline_data <-  read.csv(file.path("Data","airline_stats.csv"))

#boxplot
boxplot(pct_carrier_delay ~ airline, data = airline_data, ylim=c(0,100),las = 2)

#violin
library("ggplot2")
ggplot(data = airline_data, aes(airline, pct_carrier_delay))+
  ylim(0,100)+
  geom_violin()

File: Python/statistics/boxplot_and_violin_over_category.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

if __name__ == "__main__":
    # box plot
    file = "../data/statistics/airline_stats.csv"
    airline_stats = pd.read_csv(file)

    ax = airline_stats.boxplot(by="airline", column="pct_carrier_delay")
    plt.show()

    #violin
    ax = sns.violinplot(airline_stats.airline,
                        airline_stats.pct_carrier_delay,
                        inner="quartile",
                        color="blue")
    plt.show()